### ISSUE TITLE
[Feature Request] Unit system with dimensions displayed in meters

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#7395]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=8&t=71365

### Subproject(s) affected?

Core

### Idea description

The currently available MKS (m/kg/s/degree) unit system doesn't always result in dimensions being displayed in meters - millimeters are still used for values bellow 0.1m
For some applications (e.g. civil engineering) it would be useful to have a unit system that actually changes the display of all dimensions to meters.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by adrianinsaval** [comment]
Looking at the code MKS is supposed to display values bellow 100mm in mm and values above that but bellow 10000000mm are displayed in m and so on for even higher or lower values
https://github.com/FreeCAD/FreeCAD/blob/a8cb4e744483ce2c4b0770b948105a766330a4b2/src/Base/UnitsSchemaMKS.cpp#L57-L64
These are sane ranges IMO, if there is somewhere where mm are being used regardless of these rules I think that would be a bug but we would need examples of that.

**Comment by adrianinsaval** [comment]
@FEA-eng respond please, where is there an issue where mm are displayed with MKS? if no example is given this issue should be closed IMO, or probably closed either way and instead make an issue for the field that is not respecting the unit schema.

**Comment by FEA-eng** [comment]
@adrianinsaval I would expect the (m/kg/s/degree) unit system to display meters at all times, not only when the dimensions are close to 1 m. That's how it's done in other CAD software (like SolidWorks). Seeing mm displayed everywhere (for example the default 10 mm depth of extrusion) with MKS system enabled is really confusing. Maybe this request should be changed to something like "make the MKS system display meters regardless of the dimension value" instead.

**Comment by adrianinsaval** [comment]
I see no reason why you would want to see 10mm as 0.01m, "other software does it" is not a very good argument (although it can be valid for some stuff). Can you give me an example of a situation where forcing m for every value is advantageous?

**Comment by FEA-eng** [comment]
Small dimensions displayed in meters may seem inconvenient but it's mainly about consistency (and clarity). Especially if you are going to show the dimensions to someone, then the units should be consistent according to the rules of engineering. I'm not a civil engineer but guys from that field would likely agree with me that the MKS system should show meters at all times.

**Comment by adrianinsaval** [comment]
the problem with making MKS show in meters for values bellow 0.1m is that our default number of decimals is just 2 IIRC, meaning it'll get rounded a lot. Maybe we can make this range dependent on the configured number of decimals

**Comment by joaocoutojc** [comment]
In freecad version 0.21 rc the measurements below ten meters show in millimeters. Could be shown in millimeters measured smaller than 0.1m

**Comment by adrianinsaval** [comment]
> In freecad version 0.21 rc the measurements below ten meters show in millimeters. Could be shown in millimeters measured smaller than 0.1m

that would be when using the Standard unit system, MKS lowers this to values bellow 100mm as explained above.

**Comment by furgo16** [comment]
> Small dimensions displayed in meters may seem inconvenient but it's mainly about consistency (and clarity). Especially if you are going to show the dimensions to someone, then the units should be consistent according to the rules of engineering.

I agree with @FEA-eng's point. It is less about precision, but about consistency and predictability. I was quite confused when I set units to be `m` and measurements where displayed in `mm`. I just learned about the unwritten rule of switching to `mm` below 1 `m` measurements by trying to file an issue and stumbling upon thid one. I would argue that if someone requires more precision below 1 `m`, they are already able to switch to the other `cm` or `mm`-based unit systems.

In addition, in FreeCAD 0.22 there is a new `Meter decimal (m, m², m³)` unit system that behaves under the expectations of this issue description. That is, measurements are always shown in `m`. I would also argue that to make those two unit systems (`MKS` and `Meter decimal`) also consistent with each other, units should always be in `m`.
